middleware/kubernetes/controller.go

```diff
@@ -67,7 +70,7 @@ type dnsControl struct {
 }
 
 // newDNSController creates a controller for CoreDNS.
-func newdnsController(kubeClient *kubernetes.Clientset, resyncPeriod time.Duration, lselector *labels.Selector, initPodCache bool) *dnsControl {
+func newdnsController(kubeClient *kubernetes.Clientset, resyncPeriod time.Duration, lselector *labels.Selector, initPodCache bool, initNodeCache bool) *dnsControl {
```

Comment:
These bools needs to be backed in an Opts struct, then this function can use just an Opt parameter where these bools are fields in